bftengine/src/bftengine/messages/PrePrepareMsg.cpp

```diff
@@ -194,6 +196,29 @@ bool PrePrepareMsg::checkRequests() const {
   Assert(false);
   return false;
 }
+std::string PrePrepareMsg::getMessageClientCorrelationId(int index) {
+  auto it = RequestsIterator(this);
+  int req_num = 0;
+  while (!it.end() && req_num < index) {
+    it.gotoNext();
+    req_num++;
+  }
+  char* requestBody = nullptr;
+  it.getCurrent(requestBody);
+  ClientRequestMsg req((ClientRequestMsgHeader*)requestBody);
+  return req.getCid();
+}
+std::string PrePrepareMsg::getAllRequestsCorrelationIdAsString() {
+  std::stringstream ret;
+  ;
+  auto it = RequestsIterator(this);
+  char* requestBody = nullptr;
+  while (it.getAndGoToNext(requestBody)) {
+    ClientRequestMsg req((ClientRequestMsgHeader*)requestBody);
```

Comment:
This object can be `const`. I guess it leads to making `ClientRequestMsg::getCid()` a const member function as it is a getter.

Reply:
FIxed